rocketpool/node/collectors/beacon-collector.go

```diff
@@ -178,7 +199,35 @@ func (collector *BeaconCollector) Collect(channel chan<- prometheus.Metric) {
 		collector.upcomingSyncCommittee, prometheus.GaugeValue, upcomingSyncCommittee)
 	channel <- prometheus.MustNewConstMetric(
 		collector.upcomingProposals, prometheus.GaugeValue, upcomingProposals)
+	channel <- prometheus.MustNewConstMetric(
+		collector.recentProposals, prometheus.GaugeValue, recentProposalCount)
+}
+
+func (collector *BeaconCollector) getProposedBlockCount(validatorIndices []string, head beacon.BeaconHead, slotsPerEpoch uint64) (float64, error) {
+	// prepare for quick lookups in event of many validators:
+	indexLookup := make(map[string]string, len(validatorIndices))
+	for _, index := range validatorIndices {
+		indexLookup[index] = index
+	}
+	latestSlot := head.FinalizedEpoch*slotsPerEpoch + (slotsPerEpoch - 1)
```

Comment:
I ended up using finalized here just seemed the most straightforward and easiest to explain. Who knows baseball but us yanks anyway? :) If different values for choosing the "latest" slot are preferred here, just suggest them or commit them :)